src/request/repository/converter/request-converter.js

```diff
@@ -139,19 +140,25 @@ var getStatusCodeText = (function() {
 })();
 var setupIndex = function(request, type, payload) {
     if (type == 'begin-request') {
-        request._requestStartTime = payload.requestStartTime;
-        request._requestMethod = payload.requestMethod;
-        request._requestUrl = (payload.requestPath || '') + (payload.requestQueryString || '');
+        var url = parse(payload.url);
+        payload.path = url.pathname;
+        payload.query = url.query;
+        
+        request._requestStartTime = payload.startTime;
+        request._requestMethod = payload.method;
+        request._requestUrl = payload.path + payload.query;
         request._requestIsAjax = payload.requestIsAjax;
     }
     if (type == 'end-request') {
-        request._responseStatusCode = payload.responseStatusCode; 
-        request._responseStatusText = getStatusCodeText(payload.responseStatusCode);
-        request._responseContentType = payload.responseContentType;
-        request._responseContentCategory = getContentTypeCategory(payload.responseContentType);  
+        var contentType = payload.headers && payload.headers['Content-Type'];
```

Comment:
If I create a copy of the whole headers object, one concern is memory usage, the other being disk space. Currently we keep a copy of the last x messages the client receives in local storage. Because of this fact, we have been very careful about circular references and the data duplication/normalization. 

An example, messages are indexed via their `id`, as well as by message `type`s. The former "stores" the actual actual message and the later just as an array of `id`s by message `type. This means that if you want to get all the message objects for a given type, you need to use the array of `id`s to then lookup the messages. Even though this is a bit of a pain (which is mitigated with helper functions), it means that I can serialize the whole request object without any worries.

In this cases with headers, if we create a copy of all the headers and store that in the message or against the request, not only are the key values duplicated but the data as well. This could be mitigated in various ways, but given that the current implementation isn't really tailored towards storing too much normalized data, if we don't have any other part of the system which needs to do any programatic work on any other header, it would be adding in complexity where we don't need it atm. 

Does this make sense?